examples/preprocessing/plot_discretization_classification.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+======================
+Feature discretization
+======================
+
+A demonstration of feature discretization on synthetic classification datasets.
+Feature discretization decomposes each feature into a set of bins, here
+equally distributed in width. The discrete values are then one-hot encoded,
+and given to a linear classifier. On the two non-linearly separable datasets,
+feature discretization largely increases the performance of linear classifiers.
+
+This should be taken with a grain of salt, as the intuition conveyed by
+these examples does not necessarily carry over to real datasets.
+
+Particularly in high-dimensional spaces, data can more easily be separated
+linearly.
+
+The plots show training points in solid colors and testing points
+semi-transparent. The lower right shows the classification accuracy on the test
+set.
+"""
+print(__doc__)
+
+# Code source: Tom Dupré la Tour
+# Adapted from plot_classifier_comparison by Gaël Varoquaux and Andreas Müller
+#
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn.datasets import make_moons, make_circles, make_classification
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import KBinsDiscretizer
+from sklearn.svm import SVC, LinearSVC
+from sklearn.ensemble import GradientBoostingClassifier
+
+h = .02  # step size in the mesh
+
+
+def get_name(estimator):
+    name = estimator.__class__.__name__
+    if name == 'Pipeline':
+        name = [get_name(est[1]) for est in estimator.steps]
+        name = '\n'.join(name)
+    return name
+
+
+classifiers = [
+    (LogisticRegression(solver='lbfgs', random_state=0), {
+        'C': np.logspace(-2, 7, 10)
+    }),
+    (LinearSVC(random_state=0), {
+        'C': np.logspace(-2, 7, 10)
+    }),
+    (GradientBoostingClassifier(n_estimators=50, random_state=0), {
+        'learning_rate': np.logspace(-4, 0, 10)
+    }),
+    (SVC(random_state=0), {
+        'C': np.logspace(-2, 7, 10)
+    }),
+    (make_pipeline(
+        KBinsDiscretizer(encode='onehot'),
+        LogisticRegression(solver='lbfgs', random_state=0)), {
+            'kbinsdiscretizer__n_bins': np.arange(2, 10),
+            'logisticregression__C': np.logspace(-2, 7, 10),
+        }),
+    (make_pipeline(
+        KBinsDiscretizer(encode='onehot'), LinearSVC(random_state=0)), {
+            'kbinsdiscretizer__n_bins': np.arange(2, 10),
+            'linearsvc__C': np.logspace(-2, 7, 10),
+        }),
```

Comment:
Why are there not KBinsDiscretizer + GradientBoostingClassifier and KBinsDiscretizer + SVC? Though it seems that KBinsDiscretizer + GradientBoostingClassifier somehow performs better than GradientBoostingClassifier in the first dataset.

Reply:
Because they are non-linear classifier and the idea is to show that a linear classifier with this transformer can perform as good as a non-linear classifier, isn't it?

Reply:
Thanks :) Then I suppose it might be better to explicitly state that in the example, it seems a bit confused without your explanation at least from my side.